lib/index.js

```diff
@@ -80,6 +86,15 @@ Optimizely.prototype.track = function(track) {
   var revenue = track.revenue();
   if (revenue) payload.revenue = revenue *= 100;
   push(['trackEvent', track.event(), payload]);
+
+  var optimizelyClientInstance = window.optimizelyClientInstance;
+  if (optimizelyClientInstance && optimizelyClientInstance.track) {
+    var attributes = track.properties();
+    var userId = attributes.userId || track.userId();
```

Comment:
The precedence here should be flipped. You should first look up `track.userId()` then fallback on `attributes.userId`. However, Segment wouldn't recommend users send their userId again as properties though since that's duplicative. 

> Doing it this way because we still want the user to specify the visitor id if the know the user is not logged in.

Hmm doesn't that have duplicate profile implications? If you want to track anonymous users, you should just fallback on the Segment generated `anonymousId`s instead of asking users to add their own anonymous visitor Ids. That will also bleed into other integrations which is bad practice.

So I think this line of code should be:

```
var userId = track.userId() || track.anonymousId();
```